monkey/infection_monkey/monkey.py

```diff
@@ -230,6 +233,7 @@ def self_delete():
         if WormConfiguration.self_delete_in_cleanup \
                 and -1 == sys.executable.find('python'):
             try:
+                utils.remove_monkey_dir()
```

Comment:
ignore daniel's CR comment from before (I've talked to him). call `remove_monkey_dir` right before calling `self_delete` (like you did before).